db/migrations/00009_add_firmwares.sql

```diff
@@ -0,0 +1,24 @@
+-- +goose Up
+-- +goose StatementBegin
+
+CREATE TABLE component_firmware_version (
+  id UUID PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
+  component STRING NULL,
```

Comment:
Can all of these values actually be null? I feel like `vendor` `model` `version` and some of the other ones should require a value.

Reply:
Fair point. I made them all `NOT NULL`, recreated the SQL boiler code and updated the app code.